src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -389,41 +315,210 @@ private static <T extends Enum<T>> int mappedEnumLookup(T value, MethodHandles.L
+        List<EnumDesc<?>> enumDescs = new ArrayList<>();
+
+        byte[] classBytes = Classfile.of().build(ClassDesc.of(typeSwitchClassName(caller.lookupClass())), clb -> {
+            clb.withFlags(AccessFlag.FINAL, AccessFlag.SYNTHETIC)
+              .withMethod("typeSwitch", MethodTypeDesc.ofDescriptor("(Ljava/lang/Object;ILjava/util/function/BiFunction;)I"), Classfile.ACC_STATIC, mb -> {
+                  mb.withFlags(AccessFlag.PUBLIC, AccessFlag.FINAL, AccessFlag.STATIC)
+                    .withCode(cb -> {
+                        cb.aload(0);
+                        Label nonNullLabel = cb.newLabel();
+                        cb.if_nonnull(nonNullLabel);
+                        cb.iconst_m1();
+                        cb.ireturn();
+                        cb.labelBinding(nonNullLabel);
+                        if (labels.length == 0) {
+                            cb.constantInstruction(0)
+                              .ireturn();
+                            return ;
+                        }
+                        cb.iload(1);
+                        Label dflt = cb.newLabel();
+                        record Element(Label target, Label next, Object label) {}
+                        List<Element> cases = new ArrayList<>();
+                        List<SwitchCase> switchCases = new ArrayList<>();
+                        Object lastLabel = null;
+                        for (int idx = labels.length - 1; idx >= 0; idx--) {
+                            Object currentLabel = labels[idx];
+                            Label target = cb.newLabel();
+                            Label next;
+                            if (lastLabel == null) {
+                                next = dflt;
+                            } else if (lastLabel.equals(currentLabel)) {
+                                next = cases.getLast().next();
+                            } else {
+                                next = cases.getLast().target();
+                            }
+                            lastLabel = currentLabel;
+                            cases.add(new Element(target, next, currentLabel));
+                            switchCases.add(SwitchCase.of(idx, target));
+                        }
+                        cases = cases.reversed();
+                        switchCases = switchCases.reversed();
+                        cb.tableswitch(0, labels.length - 1, dflt, switchCases);
+                        for (int idx = 0; idx < cases.size(); idx++) {
+                            Element element = cases.get(idx);
+                            Label next = element.next();
+                            cb.labelBinding(element.target());
+                            if (element.label() instanceof Class<?> classLabel) {
+                                cb.aload(0);
+                                cb.instanceof_(classLabel.describeConstable().get());
```

Comment:
You should use orElseThrow() (see the javadoc) instead of get(). This code does not work because hidden class Class are not denotable. My first idea was that it was not a real problem because no code generated by javac will use an hidden class but this is not true. A type pattern can use the current class and this class can be loaded by a defineHiddenClass.

A way to solve that issue is to store the non denotable classes inside a list and send that list as a class data when calling defineHiddenClass.